tasks/main.yml

```diff
@@ -1,280 +1,13 @@
-    value: "{{ item }}"
-  loop: "{{ peertube_proxy_ips }}"
-  when: peertube_proxy_ips != []
-  notify:
-    - restart peertube daemon
-
-- name: Copy nginx conf file
-  template:
-    src: "nginx-peertube"
-    dest: /etc/nginx/sites-available/peertube
-    mode: 0644
-  notify:
-    - reload nginx
-
-- name: Enable nginx conf file
-  file:
-    src: /etc/nginx/sites-available/peertube
-    dest: /etc/nginx/sites-enabled/peertube
-    state: link
-  notify:
-    - reload nginx
-
-- name: Install Stretch backports repo
-  apt_repository:
-    repo: deb http://ftp.debian.org/debian stretch-backports main
-    state: present
-  when:
-    - (ansible_distribution_major_version == "9")
-  register: backports_installed
-
-- name: Change apt_preference
-  copy:
-    content: 'APT::Default-Release "stable";'
-    dest: /etc/apt/apt.conf.d/99apt_default
-    mode: 644
-  when: backports_installed is changed
-
-- name: Install certbot
-  apt:
-    pkg: python-certbot-nginx
-    state: latest
-    default_release: stretch-backports
-  register: certbot_installed
-  when: peertube_proxy_handle_https != 'yes'
-
-- name: Install Letsencrypt certificate
-  shell: |
-    certbot certonly -n \
-      --authenticator standalone \
-      --installer nginx \
-      -d {{ peertube_tld }} \
-      -m {{ admin_email }} \
-      --agree-tos \
-      --pre-hook "systemctl stop nginx" \
-      --post-hook "systemctl start nginx"
-  when:
-    - certbot_installed is changed
-    - peertube_proxy_handle_https != 'yes'
-
-- name: Insert Let's encrypt certificates in nginx
-  blockinfile:
-    dest: /etc/nginx/sites-available/peertube
-    marker: "  # {mark} let's encrypt configuration"
-    block: |2-
-        ssl_certificate /etc/letsencrypt/live/{{ peertube_tld }}/fullchain.pem;
-        ssl_certificate_key /etc/letsencrypt/live/{{ peertube_tld }}/privkey.pem;
-
-    insertbefore: ".*# Security hardening.*"
-    state: present
-  when: peertube_proxy_handle_https != 'yes'
-
-- name: Reload nginx
-  service: name=nginx state=reloaded
-
-- name: Install Peertube daemon
-  template:
-    src: peertube.service
-    dest: /etc/systemd/system/
-  notify:
-    - install peertube daemon
-
-- name: Add tools in $PATH
-  become: true
-  become_user: peertube
-  lineinfile:
-    path: "{{ peertube_user_path }}/.bashrc"
-    state: present
-    line: 'export PATH="$PATH:{{ peertube_user_path }}/peertube-latest/dist/server/tools/"'
-
-- name: Change admin password
-  become: true
-  become_user: peertube
-  shell: |-1
-    cd "{{ peertube_user_path }}/peertube-latest"
-    yarn add commander
-    # FIXME escape characters
-    echo "{{ peertube_web_admin_password }}" | NODE_CONFIG_DIR={{ peertube_user_path }}/config NODE_ENV=production npm run reset-password -- -u root
-  when:
-  - peertube_version_dir.stat.exists == False
-  - fresh_install # is changed
-  - peertube_web_admin_password is defined
+- include_tasks: copy_ldap_key_tls.yml
+  when: not peertube_ldap.insecure_tls
```

Comment:
Also check that `peertube_ldap` is defined. Now the role fails if it's undefined:

```
TASK [udelarinterior.peertube : include_tasks] ******************************************************************************************************************
task path: /home/apias/desarrolloudelar/ansible/UdelaRInterior/udelarinterior.peertube/tasks/main.yml:12
fatal: [pericon.interior.edu.uy]: FAILED! => {"msg": "The conditional check 'not peertube_ldap.insecure_tls' failed. The error was: error while evaluating conditional (not peertube_ldap.insecure_tls): 'None' has no attribute 'insecure_tls'\n\nThe error appears to be in '/home/apias/desarrolloudelar/ansible/UdelaRInterior/udelarinterior.peertube/tasks/main.yml': line 12, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- include_tasks: copy_ldap_key_tls.yml\n  ^ here\n"}


```

Reply:
Definded [here](https://github.com/UdelaRInterior/ansible-peertube/blob/724374451673d1f91aac02dd9afa9255acaa2c9e/defaults/main.yml#L68)

Reply:
Running with new changes (`peertube_ldap:[]` in defaults) I get the same error:

```
TASK [udelarinterior.peertube : include_tasks] ******************************************************************************************************************
task path: /home/apias/desarrolloudelar/ansible/UdelaRInterior/udelarinterior.peertube/tasks/main.yml:12
fatal: [pericon.interior.edu.uy]: FAILED! => {"msg": "The conditional check 'not peertube_ldap.insecure_tls' failed. The error was: error while evaluating conditional (not peertube_ldap.insecure_tls): 'list object' has no attribute 'insecure_tls'\n\nThe error appears to be in '/home/apias/desarrolloudelar/ansible/UdelaRInterior/udelarinterior.peertube/tasks/main.yml': line 12, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- include_tasks: copy_ldap_key_tls.yml\n  ^ here\n"}
```
We will have to use `is defined` condition as we talk about it.